content/general/concepts/what-is-the-factory-pattern/what-is-the-factory-pattern.md

```diff
@@ -0,0 +1,273 @@
+---
+Title: 'What is the Factory Pattern?'
+Description: 'The factory pattern defers instantiation logic of a parent abstract class to its concrete sub-classes.'
+Subjects:
+  - 'Computer Science'
+  - 'Interview Prep'
+Tags:
+  - 'Classes'
+  - 'Conceptual'
+  - 'Constructors'
+  - 'Objects'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The _factory pattern_ defers instantiation logic of a parent abstract class to its concrete sub-classes. At time of object creation, the specific class type may not be known, in which a creator class' factory method is used to decouple this identifying logic. This is typically done through the methods parameters and a switch statement.
+
+## UML Design
+
+![UML diagram of a factory](/media/factory-uml.png)
+
+## Example
+
+To illustrate the _factory pattern_, below provides a real world example, in Java, depicting potential considerations for a taxi and food delivery booking application. A user of the app will need to send a request for either a standard taxi, a large taxi, a mini bus or wish to have food delivered from a restaurant, however before the program understands their needs, which of the above requests needed remains unknown. When a request object is created, an estimated price is calculated and returned.
+
+Below provides a parent abstract class of a taxi request:
+
+```java
+public abstract class TaxiRequest<T> {
+  protected T pickUpLocation;
+  protected String destination;
+  protected double distance;
+  protected double estimatedPrice;
+
+  public TaxiRequest(final T pickUpLocation, final String destination, final int distance) {
+    this.createTaxiRequest(pickUpLocation, destination, distance);
+  }
+
+  public abstract void createTaxiRequest(final T pickUpLocation, final String destination, final int distance);
+
+
+  // getters and setters
+}
+```
+
+**Note:** As the `pickUpLocation` property may be an address or a restaurant, we have made use of generics when modeling the `TaxiRequest`. The other property's object types are known.
+
+The constructor of `TaxiRequest` calls an abstract method `createTaxtRequest`. This enforces `TaxiRequest`'s subclasses to override the method with instantiation logic. Although the constructors of the subclasses may make use of the abstract super's constructor, instantiation logic remains in the concrete.
```

Comment:
You cant remove 'in the' here as the meaning is different. We have a (in essence) 2 taxi request classes, one abstract and one concrete. The createTaxiRequest provides instantiation logic in the concrete class, and this is enforced by the same method name in the abstract class. Removing the 'in the' would mean that the method's logic is concrete. What we are saying is the method's logic lives in the concrete class (and not the abstract.)

Reply:
Could we then just append the word "class" so it reads "instantiation logic remains in the concrete class."

Thanks for the clarification! It makes much more sense that the method's logic is in the concrete class rather than the method's logic being concrete.